cwl/src/main/scala/cwl/WorkflowStep.scala

```diff
@@ -48,7 +46,7 @@ case class WorkflowStep(
 
   def fileName: Option[String] = run.select[String]
 
-  val unqualifiedStepId = WomIdentifier(Try(WorkflowStepId(id)).map(_.stepId).getOrElse(id))
+  val unqualifiedStepId = WomIdentifier(Try(FullyQualifiedName(id)).map(_.id).getOrElse(id))
```

Comment:
This seems a little smelly.... can we have an alternative `FullyQualifiedName.tryConstruct(id)` that returns the `Try` directly?

I just really don't like exceptions from constructors

Reply:
It is smelly indeed!  I'm not really sure why it's within a `Try` or why it could throw exceptions.

I would prefer to call `WorkflowStep`'s `id` field `fullyQualifiedId` or something similar but it would deviate from the spec name.  I guess we could change `FullyQualifiedName`'s `id` field to something better but I don't know what that better name is.